### PR TITLE
Creates basic API endpoints for tickets and relationship with events(/tickets, /tickets/<int:id>/events/<int:id>/tickets)

### LINKED ISSUE #3462: Implement tickets API
Parent issue #3459
To be done after #3645

Following endpoints are required
- Add ticket
- Remove ticket
- View tickets
- Edit ticket

### PR DESCRIPTION
**Pull request by magdalenesuo**
#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [X] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Implemented tickets API

#### Changes proposed in this pull request:

-Add ticket
-Remove ticket
-View tickets
-Edit ticket
-Added tickets relation with events

I had to make a rough Events api to create relationship with the tickets.

Fixes #3462

### CODE REVIEW

**Comment by niranjan94** [comment]
@shubham-padia please review

**Comment by SaptakS** [comment]
@niranjan94 @shubham-padia I have checked it locally and it works. But I am not sure about the best ways of doing it since new to rest-jsonapi library

File: app/api/tickets.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+class TicketSchema(Schema):
+
+    class Meta:
+        type_ = 'ticket'
+        self_view = 'v1.ticket_detail'
+        self_view_kwargs = {'id': '<id>'}
+
+    id = fields.Str(dump_only=True)
+    name = fields.Str()
```
Comment (niranjan94) [inline app/api/tickets.py]: Missing fields.
- sales_starts_at
- sales_ends_at
- tags
- is_hidden

File: app/api/tickets.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    description = fields.Str()
+    type = fields.Str()
+    price = fields.Float()
+    quantity = fields.Integer()
+    price = db.Column(db.Float)
```
Comment (niranjan94) [inline app/api/tickets.py]: Incorrect. Duplicate.

Comment (magdalenesuo) [inline app/api/tickets.py]: @niranjan94 this has been fixed #3706.
I'll open and issue and fix all the comment reviews along with the tax api?

Comment (niranjan94) [inline app/api/tickets.py]: Cool ! 👍

**Pull request merged**